### DISCUSSION TITLE
Hide a Menu Link based on a Count of Connections to a Table

### DISCUSSION THREAD

**Discussion by Greenlight88** [cortexrd/Knack-Toolkit-Library#222] [Q&A] [answered]
Hi all,

Hopefully screenshot below shows what I'm trying to do - would like to hide a menu link based on a count of connections

![Screenshot (326)](https://github.com/cortexrd/Knack-Toolkit-Library/assets/136812384/e1443cab-6bfc-4417-b251-0f5e45e75380)

For this example would like to hide the menu link to 'Sites' if the count of connections between company and sites is zero

Am trying to use the ktlCond as below with a jQuery - have identified the class of object as .kn-MenuLink (??) and this was my attempt

_hf=[Sites],[ktlCond, is, 0, Sites Count,Counts] $('#view_4049.kn-MenuLink:data-menu-link-count=2')

I think just something wrong with the jQuery selectorString - little help?

**Comment by cortexrd** [comment]
@Greenlight88 

I can see the confusion.  It's not very obvious when you start to use the KTL at first.

Here's what you need to add to the **Counts** details view:

`_cls=[ktllHidden], [ktlCond, is, 0, Sites Count, Counts], [ktlTarget, $('#view_4049.kn-MenuLink:data-menu-link-count=2')]`

Add class ktlHidden to the jQuery target if Sites Count has a value of zero.

Let me know if it works,
Norm

**Comment by Greenlight88** [comment]
Thank you I'm starting to get it - that didnt work though

there was a little typo in your code, but that wasnt the problem - corrected that here 

`_cls=[ktlHidden], [ktlCond, is, 0, Sites Count, Counts], [ktlTarget, $('#view_4049.kn-MenuLink:data-menu-link-count=2')]`

- I really can't see why your code wouldnt work!
I did put it in the Counts details view

heres a link to example page: https://greenlightservices.knack.com/greenlight-online#companies4/view-company2/65c41a145009110027ea2920/

**Comment by cortexrd** [comment]
I tried your application directly and found the issue.  The ktlTarget wasn't quite good.

Here's the updated string:

`_cls=[ktlHidden], [ktlCond, equ, 0, Sites Count, Counts], [ktlTarget, $('#view_4049 .knMenuLink[data-menu-link-count="2"]')]`

I also replaced **is** by **equ**, which is recommended when comparing numeric values (vs strings).

I've seen it work, so you should be good to go now.

Norm

**Comment by Greenlight88** [comment]
I still couldn'tget it tow work :(

I got this in the java console

```
KTL-0.22.19.min.js:1 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'startsWith')
    at Object.computeTargetSelector (KTL-0.22.19.min.js:1:22161)
    at KTL-0.22.19.min.js:3:58650
```

**Comment by cortexrd** [comment, marked as answer]
Hi Michael,

I could now reproduce your problem.  It is due to the keyword parser that doesn't handle the square brackets in a jQuery selector.
See issue #220 and my reply here:  https://github.com/cortexrd/Knack-Toolkit-Library/issues/220#issuecomment-1934811132

Craig has the same issue.
A fix for this may take a few days.

In the meantime, I've got another options for you.
1. you've changed the ktlCond to compare with "zero" - quotes included.  You should never add quotes.  The string must be as you see it in your app.
2. Use this jQuery instead:  **$('#view_4049 > div > nav > ul > li:nth-child(2)')**   It has no brackets and does the same job, differently.

Going back to your original version above with the "equates 0" compare, you end up with this:

`_cls=[ktlHidden], [ktlCond, equ, 0, Sites Count, Counts], [ktlTarget, $('#view_4049 > div > nav > ul > li:nth-child(2)')]`

Give it a try,
Norm

**Reply by Greenlight88** [reply]
Thanks so much Norm - I was just trying things to see if it could work - I tried making it a text string the see if the issue was the numeric field and then I thought the text field reference might need quotes and then I left it 

Understood I'll give the new code a go and if that's not up I'll keep an eye on the 220 issue

Amazing I'll be  able to do this at all really!

thank you

**Comment by Greenlight88** [comment]
Works !!

Fantastic thank you so much

**Comment by Greenlight88** [comment]
And by the by I could see this use being a useful application for people generally some variations might be 

- to make the link present but unclickable when the linked table is empty, (maybe greyed out a little) or 
- having badge numbers on the tabs representing unactioned items or similar might be super useful
- Change the wording of the tab based on summary data - eg in my example '3 x Sites'
